test/bats/plugin-test.bats

```diff
@@ -0,0 +1,225 @@
+#!/usr/bin/env bats
+
+load helpers
+
+BATS_TESTS_DIR=${BATS_TESTS_DIR:-test/bats/tests}
```

Comment:
General commnet: I see that we've added 2 new bats files. Is there any plan to have the e2e tests use these new bats files or are they purely there for external consumption?

Reply:
synced offline, I'll use the new e2e tests for testing in a separate PR.